ports/esp32/boards/sdkconfig.base

```diff
@@ -115,7 +115,7 @@ CONFIG_ADC_CAL_LUT_ENABLE=y
 CONFIG_UART_ISR_IN_IRAM=y
 
 # IDF 5 deprecated
-CONFIG_ADC_SUPPRESS_DEPRECATE_WARN=y
+CONFIG_ADC_SUPPRESS_DEPRECATE_WARN=n
```

Comment:
I think you can just remove this option, it will take the default, which is "n".